Experiment partitioning for performance spreadsheets

sadielbartholomew opened the pull request:
Update the performance template spreadsheet to enable partitions of applicable experiments to be documented separately, instead of only allowing a representative performance covering all of the appropriate experiments, where guidance is provided on the means to do this, including in a new detailed note on the front page.

sadielbartholomew committed:
Update performance spreadsheet to allow experiment partitioning

davidhassell commented:
Hi Sadie,

Thanks - a few comments:

* 1.1.5.1 and 1.1.5.2 should be seeded from the Machine spreadsheet, but should be (I realise) editable from within the performance spreadsheet - is that possible?
* Typo in Q 1.1.5.2, the reference to 1.1.4.2 should be 1.1.5.2
* I don't think we need to ask the experiments questions again for the realms. I don't think we expect (for CMIP6) a realm performance to have been noted in the absence of the overall.

sadielbartholomew committed:
Address DH feedback RE performance spreadsheets template

sadielbartholomew commented:
Thanks for your feedback @davidhassell (noting this for when you return from leave). Sorry for the delay in replying.

> 1.1.5.1 and 1.1.5.2 should be seeded from the Machine spreadsheet, but should be (I realise) editable from within the performance spreadsheet - is that possible?

Yes, that's indeed possible using the `openpyxl.worksheet.datavalidation.DataValidationList` class, as I know because it has been utilised for the seeding of the appropriate choices for the applicable experiments in the Machine spreadsheets themselves (if interested, the relevant code is mostly as follows:

https://github.com/ES-DOC/cmip6/blob/d61b741c0dfe1c49cc498858b9d7c1d3e2b547c3/lib/machines/init_xls.py#L307-L324
)

> Typo in Q 1.1.5.2, the reference to 1.1.4.2 should be 1.1.5.2

Very good spot, thanks. I've fixed it in the next commit which I am about to push up.

> I don't think we need to ask the experiments questions again for the realms. I don't think we expect (for CMIP6) a realm performance to have been noted in the absence of the overall.

Good point, I had to re-think through the possibilities to convince myself that some partitioning of the experiments in the aggregate performance wouldn't necessitate that question for each of the equivalent realm performances, but I do think you are right. I have removed the duplicate questions on the realm-specific tabs to reflect that.

sadielbartholomew commented:
I believe I have now addressed all of your feedback, @davidhassell. Given you are on leave this week, I'll merge this now to move things along, but any further feedback can be considered and addressed when I put up a PR to add the code to generate the actual per-institute and per-machine performance spreadsheets from this template.
